aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.kt

```diff
@@ -754,29 +759,58 @@ class AWSCognitoAuthPlugin : AuthPlugin<AWSCognitoAuthServiceBehavior>() {
         TODO("Not yet implemented")
     }
 
-    override fun getCurrentUser(): AuthUser? {
-        var authUser: AuthUser? = null
-        val semaphore = Semaphore(0)
+    override fun getCurrentUser(
+        onSuccess: Consumer<AuthUser>,
+        onError: Consumer<AuthException>
+    ) {
         authStateMachine.getCurrentState { authState ->
-            when (val authorizationState = authState.authNState) {
+            when (authState.authNState) {
```

Comment:
Nit: This looks like overuse of `when` structure and forming the `arrowhead` pattern making it difficult to read.
If you have only two cases, sometimes you are better off with one `if` case short circuiting.

Something like:
``` 
authStateMachine.getCurrentState { authState ->
            if (authState.authNState !is AuthenticationState.SignedIn) {
                onError.accept(AuthException.SignedOutException())
                return@getCurrentState
            }

            var token: StateChangeListenerToken? = null
             token = ...
<>
       }

Reply:
@sktimalsina Thanks Sunil. I like this pattern. Will be using this for other APIs too.